rabbitmq/resource_operator_policy.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	if err := putOperatorPolicy(rmqc, vhost, name, operatorPolicyMap); err != nil {
+		return err
+	}
+
+	id := fmt.Sprintf("%s@%s", name, vhost)
+	d.SetId(id)
+
+	return ReadOperatorPolicy(d, meta)
+}
+
+func ReadOperatorPolicy(d *schema.ResourceData, meta interface{}) error {
+	rmqc := meta.(*rabbithole.Client)
+
+	operatorPolicyId := strings.Split(d.Id(), "@")
+	if len(operatorPolicyId) < 2 {
+		return fmt.Errorf("Unable to determine operator policy ID")
+	}
+
+	name := operatorPolicyId[0]
+	vhost := operatorPolicyId[1]
+
+	operatorPolicy, err := rmqc.GetOperatorPolicy(vhost, name)
+	if err != nil {
+		return checkDeleted(d, err)
+	}
+
+	log.Printf("[DEBUG] RabbitMQ: OperatorPolicy retrieved for %s: %#v", d.Id(), operatorPolicy)
+
+	d.Set("name", operatorPolicy.Name)
+	d.Set("vhost", operatorPolicy.Vhost)
+
+	setOperatorPolicy := make([]map[string]interface{}, 1)
+	p := make(map[string]interface{})
+	p["pattern"] = operatorPolicy.Pattern
+	p["priority"] = operatorPolicy.Priority
+	p["apply_to"] = operatorPolicy.ApplyTo
+
+	operatorPolicyDefinition := make(map[string]interface{})
+	for key, value := range operatorPolicy.Definition {
+		switch v := value.(type) {
+		case float64:
+			value = strconv.FormatFloat(v, 'f', -1, 64)
+		case []interface{}:
+			var nodes []string
+			for _, node := range v {
+				if n, ok := node.(string); ok {
+					nodes = append(nodes, n)
+				}
+			}
+			value = strings.Join(nodes, ",")
+		}
+		operatorPolicyDefinition[key] = value
+	}
+	p["definition"] = operatorPolicyDefinition
+	setOperatorPolicy[0] = p
+
+	d.Set("policy", setOperatorPolicy)
+
+	return nil
+}
+
+func UpdateOperatorPolicy(d *schema.ResourceData, meta interface{}) error {
+	rmqc := meta.(*rabbithole.Client)
+
+	operatorPolicyId := strings.Split(d.Id(), "@")
+	if len(operatorPolicyId) < 2 {
+		return fmt.Errorf("Unable to determine operator policy ID")
+	}
+
+	name := operatorPolicyId[0]
+	vhost := operatorPolicyId[1]
+
+	if d.HasChange("policy") {
+		_, newOperatorPolicy := d.GetChange("policy")
+
+		operatorPolicyList := newOperatorPolicy.([]interface{})
+		operatorPolicyMap, ok := operatorPolicyList[0].(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("Unable to parse operator policy")
+		}
+
+		if err := putOperatorPolicy(rmqc, vhost, name, operatorPolicyMap); err != nil {
+			return err
+		}
+	}
+
+	return ReadOperatorPolicy(d, meta)
+}
+
+func DeleteOperatorPolicy(d *schema.ResourceData, meta interface{}) error {
+	rmqc := meta.(*rabbithole.Client)
+
+	operatorPolicyId := strings.Split(d.Id(), "@")
+	if len(operatorPolicyId) < 2 {
+		return fmt.Errorf("Unable to determine operator policy ID")
+	}
+
+	name := operatorPolicyId[0]
+	vhost := operatorPolicyId[1]
+
+	log.Printf("[DEBUG] RabbitMQ: Attempting to delete operator policy for %s", d.Id())
+
+	resp, err := rmqc.DeleteOperatorPolicy(vhost, name)
+	log.Printf("[DEBUG] RabbitMQ: OperatorPolicy delete response: %#v", resp)
+	if err != nil {
+		return err
```

Comment:
Could you just wrap all errors that come from external library please? This helps debugging error.
```suggestion
		return fmt.Errorf("could not delete operator policy: %w", err)
```